src/entity/ReputationEvent.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Entity,
+  ManyToOne,
+  PrimaryColumn,
+  Column,
+  BeforeInsert,
+  Index,
+} from 'typeorm';
+import { User } from './User';
+
+export enum ReputationReason {
+  PostUpvote = 'post_upvote',
+  PostBanned = 'post_banned',
+  CommentUpvote = 'comment_upvote',
+  PostReportConfirmed = 'post_report_confirmed',
+  SourceRequestApproved = 'source_request_approved',
+}
+
+export enum ReputationType {
+  Post = 'post',
+  Comment = 'comment',
+  Source = 'source',
+}
+
+const reputationReasonAmount: Record<ReputationReason, number> = {
+  [ReputationReason.PostUpvote]: 10,
+  [ReputationReason.PostBanned]: -100,
+  [ReputationReason.CommentUpvote]: 50,
+  [ReputationReason.PostReportConfirmed]: 100,
+  [ReputationReason.SourceRequestApproved]: 200,
+};
+
+@Entity()
+export class ReputationEvent {
```

Comment:
Make sure the length 36 is aligned with the actual primary key from the different tables, because I'm not sure about it

Reply:
Yes. User id has a max length of 36.

Reply:
What about the rest? (post id, comment id, source id)

Reply:
All 3 of them didn't specify any (default length, probably).